velox/common/memory/MemoryPool.cpp

```diff
@@ -1083,13 +1086,12 @@ void MemoryPoolImpl::setReclaimer(std::unique_ptr<MemoryReclaimer> reclaimer) {
         parent_->name());
   }
   std::lock_guard<std::mutex> l(mutex_);
-  VELOX_CHECK_NULL(reclaimer_);
-  reclaimer_ = std::move(reclaimer);
```

Comment:
@pedroerp here there's no difference, because 
1. in the old and new code `reclaimer_` is empty, so we don't need to delete it (same in both version)
2. `reclaimer` ownership transferred to `reclaimer_` (same in both version)